packages/nocodb/src/lib/version-upgrader/ncFilterUpgrader_0104004.ts

```diff
@@ -0,0 +1,370 @@
+      await Filter.update(
+        filter.id,
+        {
+          comparison_op: 'nanyof',
+          value: newFilterValue,
+        },
+        ncMeta
+      )
+    );
+  }
+  return actions;
+};
+
+const migrateToCheckboxFilter = async (filter, actions: any[], ncMeta) => {
+  if (['empty', 'null'].includes(filter.comparison_op)) {
+    // migrate to not checked
+    actions.push(
+      await Filter.update(
+        filter.id,
+        {
+          comparison_op: 'notchecked',
+        },
+        ncMeta
+      )
+    );
+  } else if (['notempty', 'notnull'].includes(filter.comparison_op)) {
+    //  migrate to checked
+    actions.push(
+      await Filter.update(
+        filter.id,
+        {
+          comparison_op: 'checked',
+        },
+        ncMeta
+      )
+    );
+  } else if (filter.comparison_op === 'eq') {
+    if (['true', 'True', '1', 'T', 'Y'].includes(filter.value)) {
+      // migrate to checked
+      actions.push(
+        await Filter.update(
+          filter.id,
+          {
+            comparison_op: 'checked',
+            value: '',
+          },
+          ncMeta
+        )
+      );
+    } else if (['false', 'False', '0', 'F', 'N'].includes(filter.value)) {
+      // migrate to notchecked
+      actions.push(
+        await Filter.update(
+          filter.id,
+          {
+            comparison_op: 'notchecked',
+            value: '',
+          },
+          ncMeta
+        )
+      );
+    } else {
+      // invalid value - good to delete
+      actions.push(await Filter.delete(filter.id, ncMeta));
+    }
+  } else if (filter.comparison_op === 'neq') {
+    if (['false', 'False', '0', 'F', 'N'].includes(filter.value)) {
+      // migrate to checked
+      actions.push(
+        await Filter.update(
+          filter.id,
+          {
+            comparison_op: 'checked',
+            value: '',
+          },
+          ncMeta
+        )
+      );
+    } else if (['true', 'True', '1', 'T', 'Y'].includes(filter.value)) {
+      // migrate to not checked
+      actions.push(
+        await Filter.update(
+          filter.id,
+          {
+            comparison_op: 'notchecked',
+            value: '',
+          },
+          ncMeta
+        )
+      );
+    } else {
+      // invalid value - good to delete
+      actions.push(await Filter.delete(filter.id, ncMeta));
+    }
+  }
+  return actions;
+};
+
+async function migrateFilters(ncMeta: NcMetaIO) {
+  const filters = await ncMeta.metaList2(null, null, MetaTable.FILTER_EXP);
+  let actions = [];
+  for (const filter of filters) {
+    const col = await Column.get({ colId: filter.fk_column_id }, ncMeta);
+    if (!col || !col.uidt) {
+      // column not found -> skip
+      continue;
+    }
+    if (
+      [
+        UITypes.SingleLineText,
+        UITypes.LongText,
+        UITypes.PhoneNumber,
+        UITypes.Email,
+        UITypes.URL,
+      ].includes(col.uidt)
+    ) {
+      actions = await removeArithmeticFilters(filter, actions, ncMeta);
```

Comment:
You don't have to assign the return value since it's the same array.

Instead of passing `actions` and mutating you can return a new array from the function and push it to `actions`.